app/policies/user_policy.rb

```diff
@@ -33,4 +33,8 @@ def update?
   def destroy?
     @current_user.superadmin? && !@user.superadmin?
   end
+
+  def reset_password?
+    @current_user.superadmin? && @current_user != @user
```

Comment:
Nope. The user himself can do this within the login form with devises own password forgotten feature.